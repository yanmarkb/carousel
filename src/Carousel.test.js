import { render, fireEvent } from "@testing-library/react";
import Carousel from "./Carousel";
import TEST_IMAGES from "./_testCommon.js";

it("works when you click on the right arrow", function () {
	const { container } = render(
		<Carousel photos={TEST_IMAGES} title="images for testing" />
	);
	// expect the first image to show, but not the second
	expect(
		container.querySelector('img[alt="testing image 1"]')
	).toBeInTheDocument();
	expect(
		container.querySelector('img[alt="testing image 2"]')
	).not.toBeInTheDocument();

	// move forward in the carousel
	const rightArrow = container.querySelector(".bi-arrow-right-circle");
	fireEvent.click(rightArrow);

	// expect the second image to show, but not the first
	expect(
		container.querySelector('img[alt="testing image 1"]')
	).not.toBeInTheDocument();
	expect(
		container.querySelector('img[alt="testing image 2"]')
	).toBeInTheDocument();
});

//Smoke test
test("renders Carousel without crashing", () => {
	const photos = [
		{ src: "https://example.com/image1.jpg", caption: "Image 1" },
		{ src: "https://example.com/image2.jpg", caption: "Image 2" },
	];
	const title = "Test Carousel";
	render(<Carousel photos={photos} title={title} />);
});

//Snapshot test
test("matches snapshot", () => {
	const photos = [
		{ src: "https://example.com/image1.jpg", caption: "Image 1" },
		{ src: "https://example.com/image2.jpg", caption: "Image 2" },
	];
	const title = "Test Carousel";
	const { asFragment } = render(<Carousel photos={photos} title={title} />);
	expect(asFragment()).toMatchSnapshot();
});

test("left arrow moves to the previous image", () => {
	const photos = [
		{ src: "https://example.com/image1.jpg", caption: "Image 1" },
		{ src: "https://example.com/image2.jpg", caption: "Image 2" },
	];
	const title = "Test Carousel";
	const { getByLabelText, rerender, getByAltText } = render(
		<Carousel photos={photos} title={title} />
	);

	// Move to the second image
	fireEvent.click(getByLabelText("Go forward"));

	// Click the left arrow
	fireEvent.click(getByLabelText("Go back"));

	// Re-render the component to get the updated state
	rerender(<Carousel photos={photos} title={title} />);

	// Check that the first image is displayed
	expect(getByAltText("Image 1")).toBeInTheDocument();
});

test("left arrow is missing on the first image and right arrow is missing on the last image", () => {
	const photos = [
		{ src: "https://example.com/image1.jpg", caption: "Image 1" },
		{ src: "https://example.com/image2.jpg", caption: "Image 2" },
	];
	const title = "Test Carousel";
	const { getByLabelText, queryByLabelText, rerender } = render(
		<Carousel photos={photos} title={title} />
	);

	// Check that the left arrow is not in the document
	expect(queryByLabelText("Go back")).not.toBeInTheDocument();

	// Move to the second image
	fireEvent.click(getByLabelText("Go forward"));

	// Re-render the component to get the updated state
	rerender(<Carousel photos={photos} title={title} />);

	// Check that the right arrow is not in the document
	expect(queryByLabelText("Go forward")).not.toBeInTheDocument();
});
