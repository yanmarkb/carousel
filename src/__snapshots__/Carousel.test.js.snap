// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Carousel"
  >
    <h1>
      Test Carousel
    </h1>
    <div
      class="Carousel-main"
    >
      <div
        class="Card"
      >
        <h4
          class="Card-title"
        >
          Image 1
        </h4>
        <img
          alt="Image 1"
          class="Card-image"
          src="https://example.com/image1.jpg"
        />
        <small
          class="Card-small"
        >
          Image 1 of 2.
        </small>
      </div>
      <i
        aria-label="Go forward"
        class="bi bi-arrow-right-circle"
      />
    </div>
  </div>
</DocumentFragment>
`;
